Add SoC SXR2250P

cccclai opened the pull request:
Differential Revision: D69063201

cccclai commented:
I may need some help to figure out the exact config for the SoC SXR2250P..

winskuo-quic commented on backends/qualcomm/serialization/qc_schema.py:
```diff
@@ -60,6 +61,7 @@ class SocInfo:
     QcomChipset.SM8750: SocInfo(QcomChipset.SM8750, HtpInfo(HtpArch.V79, 8)),
     QcomChipset.SSG2115P: SocInfo(QcomChipset.SSG2115P, HtpInfo(HtpArch.V73, 2)),
     QcomChipset.SA8295: SocInfo(QcomChipset.SA8295, HtpInfo(HtpArch.V68, 8)),
+    QcomChipset.SXR2250P: SocInfo(QcomChipset.SXR2250P, HtpInfo(HtpArch.V68, 8)),
```
This should be v69

haowhsu-quic replied:
`HtpArch.V69`, `8`

winskuo-quic commented on backends/qualcomm/serialization/qc_compiler_spec.fbs:
```diff
@@ -39,6 +39,7 @@ enum QcomChipset: int {
   SM8650 = 57,
   SA8295 = 39,
   SM8750 = 69
+  SXR2250P = 36,
```
Could you try 53? Thanks

winskuo-quic commented on backends/qualcomm/serialization/qc_schema.py:
```diff
@@ -44,6 +44,7 @@ class QcomChipset(IntEnum):
     SM8650 = 57  # v75
     SA8295 = 39  # v68
     SM8750 = 69  # v79
+    SXR2250P = 42  # v69
```
Also 53 for this.

haowhsu-quic replied:
I think it should be 53.

cccclai commented:
@winskuo-quic I added a few more SoCs to support, but not sure some of the specs. Can I get some help with this?

winskuo-quic commented on backends/qualcomm/serialization/qc_compiler_spec.fbs:
```diff
@@ -38,7 +38,11 @@ enum QcomChipset: int {
+  SXR1230P = x,
+  SXR2230P = x,
+  SXR2250P = 53
+  SSG2125P = x
```
SXR1230P = 45,
SXR2230P = 53,
SSG2125P = 58,

SXR2230P and SXR2250P are both 53. I think we could keep SXR2230P = 53 here and remove SXR2550P since SXR2550P is a variant of SXR2230P

winskuo-quic commented on backends/qualcomm/serialization/qc_schema.py:
```diff
@@ -44,7 +44,11 @@ class QcomChipset(IntEnum):
+    SXR2230P = 1 # place holder
+    SSG2115P = 1 # place holder
+    SSG2125P = 1 # place holder
+    SXR2250P = 53  # v69
```
SXR1230P = 45,
SXR2230P = 53,
SSG2125P = 58,

SSG2115P is duplicate
Also consider remove SXR2250P and keep SXR2230P = 53 due to the same reason above

winskuo-quic commented on backends/qualcomm/serialization/qc_schema.py:
```diff
@@ -60,6 +64,11 @@ class SocInfo:
+    QcomChipset.SXR2230P: SocInfo(QcomChipset.SXR2230P, HtpInfo(HtpArch.V69, 8)),
+    QcomChipset.SSG2125P: SocInfo(QcomChipset.SSG2125P, HtpInfo(HtpArch.V69, 8)),
+    QcomChipset.SXR2250P: SocInfo(QcomChipset.SXR2250P, HtpInfo(HtpArch.V69, 8)),
 }
```
QcomChipset.SXR2250P: SocInfo(QcomChipset.SXR2250P, HtpInfo(HtpArch.V69, 8)),
QcomChipset.SXR1230P: SocInfo(QcomChipset.SXR1230P, HtpInfo(HtpArch.V73, 2)),
QcomChipset.SXR2230P: SocInfo(QcomChipset.SXR2230P, HtpInfo(HtpArch.V69, 8)),
QcomChipset.SSG2125P: SocInfo(QcomChipset.SSG2125P, HtpInfo(HtpArch.V73, 2)),


QcomChipset.SXR2250P is dupicate.
Also consider remove SXR2250P and keep SXR2230P due to the same reason above.

winskuo-quic reviewed (commented):
Hi @cccclai,
I think SXR2250P is a variant of SXR2230P, and their DspArch and VTCM size should be the same. Could you try it on your side and see if compiling SXR2250P using `-m SXR2230P` works?
Thanks.

cccclai commented:
@winskuo-quic yes it works! I use `SXR2230P` and model add.pte can be executed
```
I 00:00:00.000172 executorch:qnn_executor_runner.cpp:160] Model file add.pte is loaded.
I 00:00:00.000201 executorch:qnn_executor_runner.cpp:170] Using method forward
I 00:00:00.000210 executorch:qnn_executor_runner.cpp:217] Setting up planned buffer 0, size 48.
[INFO] [Qnn ExecuTorch]: Deserializing processed data using QnnContextCustomProtocol
[INFO] [Qnn ExecuTorch]: create QNN Logger with log_level 2
[WARNING] [Qnn ExecuTorch]:  <W> Initializing HtpProvider

[WARNING] [Qnn ExecuTorch]:  <W> Function not called, PrepareLib isn't loaded!

[INFO] [Qnn ExecuTorch]: Initialize Qnn backend parameters for Qnn executorch backend type 2
[INFO] [Qnn ExecuTorch]: Caching: Caching is in RESTORE MODE.
[INFO] [Qnn ExecuTorch]: QnnContextCustomProtocol expected magic number: 0x5678abcd but get: 0x2000000
[WARNING] [Qnn ExecuTorch]:  <W> Specified config SOC, ignoring on real target

[WARNING] [Qnn ExecuTorch]:  <W> Function not called, PrepareLib isn't loaded!

[WARNING] [Qnn ExecuTorch]:  <W> Function not called, PrepareLib isn't loaded!

[INFO] [Qnn ExecuTorch]: Running level=3 optimization.
I 00:00:00.134639 executorch:qnn_executor_runner.cpp:241] Method loaded.
E 00:00:00.134655 executorch:method.cpp:1056] Output 0 is memory planned, or is a constant. Cannot override the existing data pointer.
I 00:00:00.134665 executorch:qnn_executor_runner.cpp:301] ignoring error from set_output_data_ptr(): 0x2
I 00:00:00.134669 executorch:qnn_executor_runner.cpp:304] Inputs prepared.
I 00:00:00.134709 executorch:qnn_executor_runner.cpp:447] Input list not provided. Inputs prepared with default values set.
I 00:00:00.135125 executorch:qnn_executor_runner.cpp:454] Model executed successfully.
[INFO] [Qnn ExecuTorch]: Destroy Qnn backend parameters
[INFO] [Qnn ExecuTorch]: Destroy Qnn context
[INFO] [Qnn ExecuTorch]: Destroy Qnn device
```

cccclai commented:
@winskuo-quic if this PR looks good to you, mind approving it?

winskuo-quic reviewed (approved):
LGTM. Thanks

winskuo-quic commented:
> @winskuo-quic if this PR looks good to you, mind approving it?

Sorry. Forgot to press approve when leaving the comments.

cccclai committed:
Add SoC SXR1230P, SXR2230P, SSG2125P (#8148)

Summary:


As title, need to support SoC SXR2250P

Differential Revision: D69063201
